src/procstat.c

```diff
@@ -394,8 +422,18 @@ static void fuse_readdir(fuse_req_t req, fuse_ino_t ino, size_t size, off_t off,
 
 static bool allowed_open(struct procstat_item *item, struct fuse_file_info *fi)
 {
+	struct procstat_file *file;
 	/* non control items can be only opened as readonly */
-	return (((fi->flags & 3) == O_RDONLY || item_type_control(item)));
+	if ((fi->flags & 3) == O_RDONLY)
+		return true;
+
+	if (item_type_control(item))
+		return true;
+
+	file = container_of(item, struct procstat_file, base);
+	if (file->writer)
```

Comment:
at this point you assume that the file is writeable?

Reply:
yes .. when use "passes" writer it makes "property" writable